elementary/files
# Fix single click activate

Fixes #292 

The above issues were found to be due to delayed (throttled) updating of selected files list due to performance issues with large selections.  This PR refactors selection handling in the view so that the selection list and menu are only updated on demand (if necessary) - selection changed signals only mark it as "invalid".

The selected list handling is now solely the responsibility of AbstractDirectoryView for simplicity, and consequent changes were made to derived classes.  The selected files list now contains unowned references to model objects to ensure they are not inadvertantly destroyed..

TO TEST:
* Check that, in single-click mode, the issues are fixed
* Check that no other change in behaviour involving selections (activation, DND, copy/cut/paste etc) occurs.
* Check that selecting and copying or moving large numbers of files to a new folder is not slowed down.

See https://github.com/jeremypw/elementary-files-testing for suggested test cases.
